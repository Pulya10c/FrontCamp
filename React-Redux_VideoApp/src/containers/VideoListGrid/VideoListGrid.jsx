import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import Grid from "components/Grid";
import Card from "components/Card";
import { moviesDataSelector } from "selectors/videoListSelectors";
import { PATHS } from "src/constants";

const mapStateToProps = state => ({
  movies: moviesDataSelector(state)
});

const VideoListGrid = ({ movies }) => {
  // static propTypes = {
  //   prop: PropTypes
  // };

  return (
    <Grid>
      {movies.length
        ? movies.map(({ id, title, release_date, poster_path, genres }) => (
            <Card
              key={id}
              to={`/film/${id}`}
              posterPath={poster_path}
              title={title}
              description={genres.join(" ")}
              year={release_date}
            />
          ))
        : "No films found"}
    </Grid>
  );
};

VideoListGrid.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string,
  release_date: PropTypes.string,
  poster_path: PropTypes.string,
  genres: PropTypes.array
};

export default connect(mapStateToProps, null)(VideoListGrid);
