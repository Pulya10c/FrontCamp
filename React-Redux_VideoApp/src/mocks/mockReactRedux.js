const createMockMapStateToProps = mapStateToProps => {
  if (!mapStateToProps) {
    return mapStateToProps;
  }

  return (state, ownProps) => {
    return {
      props: mapStateToProps(state, ownProps)
    };
  };
};

const getDispatchFunctionFormShorthandObject = actionsObj => dispatch => {
  return Object.keys(actionsObj).reduce((props, actionKey) => {
    return {
      ...props,
      [actionKey]: function() {
        return dispatch(actionsObj[actionKey].apply(this, arguments));
      }
    };
  }, {});
};

const createMockMapDispatchToProps = mapDispatchToProps => {
  if (!mapDispatchToProps) {
    return mapDispatchToProps;
  }

  const mappingFunction =
    typeof mapDispatchToProps === 'function'
      ? mapDispatchToProps
      : getDispatchFunctionFormShorthandObject(mapDispatchToProps);

  return (dispatch, ownProps) => {
    dispatch = dispatch || jest.fn(action => action);

    return {
      props: mappingFunction(dispatch, ownProps),
      dispatch
    };
  };
};

const createMockMergeProps = mergeProps => {
  if (!mergeProps) {
    return mergeProps;
  }

  return (stateProps, dispatchProps, ownProps) => {
    dispatchProps = dispatchProps || { dispatch: jest.fn(action => action) };

    return {
      props: mergeProps(stateProps, dispatchProps, ownProps),
      dispatch: dispatchProps.dispatch
    };
  };
};

export default {
  connect: (
    mapStateToProps,
    mapDispatchToProps,
    mergeProps
  ) => ConnectedComponent => ({
    mapStateToProps: createMockMapStateToProps(mapStateToProps),
    mapDispatchToProps: createMockMapDispatchToProps(mapDispatchToProps),
    mergeProps: createMockMergeProps(mergeProps),
    ConnectedComponent
  })
};
