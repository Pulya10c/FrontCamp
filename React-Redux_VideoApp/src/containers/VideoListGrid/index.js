import VideoListGrid from './VideoListGrid';

export default VideoListGrid;
