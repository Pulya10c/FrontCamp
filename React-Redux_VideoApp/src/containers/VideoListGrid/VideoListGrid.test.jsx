import React from "react";
import { shallow } from "enzyme";
import mockReactRedux from "mocks/mockReactRedux";
import VideoListGrid from "./";

jest.mock("react-redux", () => mockReactRedux);

const mockMovies = [
  {
    id: "mockId",
    title: "mockTitle",
    release_date: "mockRelease_date",
    poster_path: "mockPoster_path",
    genres: ["mockGenre"]
  }
];

jest.mock("selectors/videoListSelectors", () => ({
  moviesDataSelector: jest.fn(() => mockMovies)
}));

describe("Component VideoListGrid", () => {
  const { mapStateToProps, ConnectedComponent } = VideoListGrid;

  it("should render correctly", () => {
    const { props } = mapStateToProps();
    expect(shallow(<ConnectedComponent {...props} />)).toMatchSnapshot();
  });

  it("should pass correct props to component from mapStateToProps", () => {
    const { props } = mapStateToProps();
    expect(props).toEqual({
      movies: mockMovies
    });
  });
});
