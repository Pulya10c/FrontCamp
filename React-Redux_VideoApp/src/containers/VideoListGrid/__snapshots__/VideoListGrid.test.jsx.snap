// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component VideoListGrid should render correctly 1`] = `
<styled.section>
  <Card
    description="mockGenre"
    key="mockId"
    posterPath="mockPoster_path"
    title="mockTitle"
    to="/film/mockId"
    year="mockRelease_date"
  />
</styled.section>
`;
